### DISCUSSION TITLE
Non-rectilinear core area

### DISCUSSION THREAD

**Discussion by Blebowski** [The-OpenROAD-Project/OpenROAD#5199] [General]
Hello,

is it possible to somehow specify core-area that is non-rectilinear (e.g. as a group of polygons surrounding concave shape) ?

I understand that if doing "digital on top" FP where all the analog is a digital macro, this does not make sense. However, it may be usefull in case where "analog-on-top" is done, and digital may have some weird shape and is eventually placed as hard GDSII macro into analog-on-top design.

**Comment by maliberty** [comment]
The simplest would be to make a rectangular block and add placement and routing blockages for the disallowed area

**Comment by Blebowski** [comment]
Thanks. Can I then do the `add_pdn_ring` around just the cut area (e.g. L-shape pdn-ring) ?  Or is there some other trick how
to achieve this ?

**Reply by maliberty** [reply]
I've never heard of a non rectangular pad ring. Do you have a chip that is not a rectangle??

**Reply by Blebowski** [reply]
No. What I meant, was having only power-ring around the core area in non-rectilinear shape.
E.g. L-shaped core, with logic power rails ring around its perimeter (also L-shaped).
I don't have particular use-case, I am just trying to learn the tool...

I remember that I have seen PnR guys fitting logic into very weirdly shaped core areas with lot of
analog around (in analog on top approach). This was in some commercial tool. Digital logic was
then sort-of macro that was fitted in chip-top assembly by the analog guys. Since the logic was
non-rectangular shaped, the logic power ring was following the perimeter of the logic.

So, I am just trying to re-create similar scenarios as I have seen back-end guys dealing with in
my former jobs as front-end designer.

I hope its not too stupid or daring question. I took the advice of Matt Ven from Efabless movement:
"Be new at something and be bad at it". :)

**Reply by maliberty** [reply]
We didn't have much demand for such a feature so the support is incomplete. It might work but I suppose you will hit problems.
